File: norintegrate-mcp/src/main/java/com/norintegrate/mcp/tool/ProcedureDetailTool.java
package com.norintegrate.mcp.tool;

import com.norintegrate.common.procedure.ProcedureService;
import org.springframework.ai.tool.annotation.Tool;
import org.springframework.ai.tool.annotation.ToolParam;
import org.springframework.stereotype.Component;

@Component
public class ProcedureDetailTool {

  private final ProcedureService procedureService;

  public ProcedureDetailTool(ProcedureService procedureService) {
    this.procedureService = procedureService;
  }

  @Tool(
      name = "getProcedureDetail",
      description =
          "Get detailed information about a specific settlement procedure including required"
              + " documents, responsible authority, and estimated processing time")
  public ProcedureDetailResult getProcedureDetail(
      @ToolParam(description = "The numeric ID of the procedure to look up") long procedureId) {
    var procedure = procedureService.findById(procedureId);
    var requirements = procedureService.getDocumentRequirements(procedureId);
    var documents =
        requirements.stream()
            .map(
                req ->
                    new DocumentItem(
                        req.getDocumentName(), req.getDescription(), req.isMandatory()))
            .toList();
    return new ProcedureDetailResult(
        procedure.getId(),
        procedure.getTitle(),
        procedure.getDescription(),
        procedure.getAuthority(),
        procedure.getEstimatedDays(),
        documents);
  }
}

File: norintegrate-mcp/src/main/kotlin/com/norintegrate/mcp/tool/ProcedureDetailTool.kt
package com.norintegrate.mcp.tool

import com.norintegrate.common.procedure.ProcedureService
import org.springframework.ai.tool.annotation.Tool
import org.springframework.ai.tool.annotation.ToolParam
import org.springframework.stereotype.Component

@Component
class ProcedureDetailTool(
    private val procedureService: ProcedureService,
) {
    @Tool(
        name = "getProcedureDetail",
        description =
            "Get detailed information about a specific settlement procedure including required" +
                " documents, responsible authority, and estimated processing time",
    )
    fun getProcedureDetail(
        @ToolParam(description = "The numeric ID of the procedure to look up") procedureId: Long,
    ): ProcedureDetailResult {
        val procedure = procedureService.findById(procedureId)
        val documents =
            procedureService.getDocumentRequirements(procedureId).map { req ->
                DocumentItem(req.documentName, req.description, req.isMandatory)
            }
        return ProcedureDetailResult(
            procedure.id,
            procedure.title,
            procedure.description,
            procedure.authority,
            procedure.estimatedDays,
            documents,
        )
    }
}
